src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerations/AvoidMultipleEnumerations.cs

```diff
@@ -138,37 +145,60 @@ public override void Initialize(AnalysisContext context)
             context.RegisterCompilationStartAction(CompilationStartAction);
         }
 
-        private void CompilationStartAction(CompilationStartAnalysisContext context)
-            => context.RegisterOperationBlockStartAction(OnOperationBlockStart);
-
-        private void OnOperationBlockStart(OperationBlockStartAnalysisContext operationBlockStartAnalysisContext)
+        private void CompilationStartAction(CompilationStartAnalysisContext compilationStartAnalysisContext)
         {
-            var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(operationBlockStartAnalysisContext.Compilation);
-            var deferredMethods = GetLinqMethods(wellKnownTypeProvider, s_deferParametersEnumeratedLinqMethods);
+            var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+            var linqChainMethods = GetLinqMethods(wellKnownTypeProvider, s_linqChainMethods);
+            var noEnumerationMethods = GetLinqMethods(wellKnownTypeProvider, s_noEnumerationLinqMethods);
             var enumeratedMethods = GetEnumeratedMethods(wellKnownTypeProvider, s_immutableCollectionsTypeNamesAndConvensionMethods, s_enumeratedParametersLinqMethods);
             var noEffectLinqChainMethods = GetLinqMethods(wellKnownTypeProvider, s_noEffectLinqChainMethods);
-            var compilation = operationBlockStartAnalysisContext.Compilation;
+            var compilation = compilationStartAnalysisContext.Compilation;
             var additionalDeferTypes = GetTypes(compilation, s_additionalDeferredTypes);
-
             // In CFG blocks there is no foreach loop related Operation, so use the
             // the GetEnumerator method to find the foreach loop
             var getEnumeratorSymbols = GetGetEnumeratorMethods(wellKnownTypeProvider);
+
+            var syntaxTree = compilation.SyntaxTrees.FirstOrDefault();
+            var customizedNoEnumerationMethods = syntaxTree == null
```

Comment:
This doesn't seem right, we should compute these option values inside OperationBlockStart using the syntax tree associated with the operation block. These options come from editorconfig file, which can be defined at any folder/file level and need to be same across the compilation.

Reply:
Note that we have perf optimization in place in our options computation infrastructure, so if the underlying dictionary for the editorconfig options is same for multiple syntax trees across compilation, computing option values for each syntax tree will only get the perf hit for the first syntax tree.

Reply:
OK I will revert that commit